src/controllers/commands_controller.js

```diff
@@ -0,0 +1,207 @@
+/**
+ * Commands Controller.
+ *
+ * Manages HTTP requests to /commands.
+ *
+ * Grammar that the parser understands:
+ *  Turn the <tag> light <on|off>
+ *  Turn <tag> <on|off>
+ *  Shut <tag> <on|off>
+ *  Shut the <tag> light <on|off>
+ *
+ * <tag> must match the .name property of one of the /things objects
+ *  in order for the command to be executed.
+ *
+ * Sample curl command to test from the command line:
+ * curl -H 'Authorization:Bearer '<JSONWebToken>'
+ *        -H "Content-Type: application/json"
+ *        -H "Accept: application/json"
+ *        -d '{"text":"turn the kitchen light on"}'
+ *        https://localhost:4443/commands
+ *        -k
+ *
+ * HTTP Response Codes        Status message
+ *              201           Command Created
+ *              400           Text element not defined
+ *              404           Thing Not Found
+ *              406           Unable to understand command
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+'use strict';
+
+const express = require('express');
+const Constants = require('../constants.js');
+const rp = require('request-promise');
+
+const CommandsController = express.Router();
+
+
+const aiOptions = {
+  uri: 'https://api.api.ai/api/query',
+  method: 'POST',
+  body: '',
+  headers: {
+    'Authorization': 'Bearer af2bccd942c24fd68622efc6b4c8526c',
+    'Content-Type': 'application/json'
+  }
+};
+
+const thingsOptions = {
+  uri: '',
+  rejectUnauthorized: false,
+  body: '',
+  method: 'GET',
+  headers: {
+    'Authorization': '',
+    'Accept': 'application/json'
+  }
+};
+
+const iotOptions = {
+  uri: '',
+  method: 'PUT',
+  rejectUnauthorized: false,
+  headers: {
+    'Accept': 'application/json',
+    'Content-Type': 'application/json'
+  },
+  body: ''
+};
+
+/**
+ * Local Variables for the Gateway Href and the Web Token since the
+ * CommandsController will be posting to itself.
+ */
+CommandsController.gatewayRef = '';
+CommandsController.jwt = '';
+
+/**
+ * Called by the app.js to configure the auth header and the address of the GW
+ *  since the CommandsController will be posting to itself.
+ */
+CommandsController.configure =  function(gatewayHref, jwt) {
+  CommandsController.gatewayHref = gatewayHref;
+  CommandsController.jwt = jwt;
+};
+
+/**
+ * Parses the intent for a text sentence and sends to the API.ai intent
+ * parser to determine intent.  Then executes the intent as an action on the
+ * thing API.
+ */
+CommandsController.post('/', function (request, response) {
+
+  if(!request.body || request.body['text'] === undefined) {
+    response.status(400).send('Text element not defined');
```

Comment:
As an API endpoint, errors should be sent back in a consistent JSON format like `{error: 'Message goes here'}`.

Reply:
Added these as a message in JSON format.